api/v1alpha1/repository_types.go

```diff
@@ -28,14 +51,38 @@ type RepositorySpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of Repository. Edit repository_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// URL chart repository address
+	// +kubebuilder:validation:Required
+	URL string `json:"url"`
+
+	// BasicAuthSecret if the chart repository requires basic auth authentication,
+	// set the username and password to secret, with the fields user and password respectively.
+	BasicAuthSecret string `json:"basicAuthSecret,omitempty"`
```

Comment:
```go
// RepositoryAuth defines the authentication information for repository
// Inspired by https://github.com/helm/helm/blob/v3.12.0/cmd/helm/registry_login.go#L38
type RepositoryAuth struct {
	username string
	password string
	certFile string
	keyFile  string
	caFile   string
	insecure bool
}
```

Reply:
We can use `RepositoryAuth ` to define aut details, but save it into a k8s secret to avoid leaking credentails.

Reply:
> save it into a k8s secret

I agree with this.

And I think `insecure` should in spec, I think in many cases, the internal helm package management platform does not require authentication and it is insecure

Reply:
make sense to me